includes/payment-methods/class-upe-payment-gateway.php

```diff
@@ -285,6 +287,56 @@ public function process_redirect_payment( $order_id, $intent_id, $save_payment_m
 		}
 	}
 
+	/**
+	 * Set formatted readable payment method title for order,
+	 * using payment method details from accompanying charge.
+	 *
+	 * @param WC_Order $order WC Order being processed.
+	 * @param array    $payment_method_details Array of payment method details from charge.
```

Comment:
You can find a description of the various shapes this `payment_method_details` can take over [here](https://stripe.com/docs/api/charges/object#charge_object-payment_method_details).